### DISCUSSION TITLE
course/06

### DISCUSSION THREAD

**Discussion by giscus** [mitre/inspec-advanced-developer#4] [Ideas]
# course/06

The MITRE Security Automation Framework Team's Advanced InSpec profile development course

https://deploy-preview-2--mitre-inspec-advanced-developer.netlify.app/course/06.html

**Comment by aaronlippold** [comment]
make sure the 'input commands' are a solo block so they are easy to copy and paste

**Comment by karikarshivani** [comment]
This might be a good section to show the students what's under the hood for the resource(s) that they worked with during the basic course. Tying the concepts of the `name` attribute in the class becoming the resource name, arguments for that resource being set with `initialize` and the class methods that are later used with `its` in the describe block - would lead to a smoother transition.

**Comment by Rlin232** [comment]
Suggestion to clarify the application/role of resources: Maybe add a line that says something similar to "they wrap around a given service and provide an api that allows you to ask questions regarding the configuration"
